packages/web3-eth/src/accounts.js

```diff
@@ -111,11 +111,99 @@ function recover (hash, signature) {
     return EthFP.Account.recover(hash, signature);
 }
 
+function encrypt (account, password) {
+    // TODO: implement
+    return JSON.stringify(account);
+}
+
+function decrypt (account, password) {
+    return JSON.parse(account);
+}
+
+// Note: this is trying to follow closely the specs on
+// http://web3js.readthedocs.io/en/1.0/web3-eth-accounts.html
+
+function Wallet() {
+    this.length = 0;
+}
+
+Wallet.prototype.create = function (numberOfAccounts, entropy) {
+    for (var i = 0; i < numberOfAccounts; ++i) {
+        this.add(create(entropy).privateKey);
+    }
+    return this;
+};
+
+Wallet.prototype.add = function (account) {
+    if (typeof account === "string") {
+        account = privateToAccount(account);
+    }
+    if (!this[account.address]) {
+        this[this.length++] = account;
+        this[account.address] = account;
+    }
+};
+
+Wallet.prototype.remove = function (addressOrIndex) {
+    var account = this[addressOrIndex];
+    for (var i = 0; i < this.length; ++i) {
+        if (this[i] && this[i].address === account.address) {
+            delete this[account.address];
+            this[i].address = null;
+            this[i].privateKey = null;
+            delete this[i];
+        }
+        if (!this[i]) {
+            if (i < this.length - 1) {
+                this[i] = this[i + 1];
+                delete this[i + 1];
+            } else {
+                --this.length;
+            }
+        }
+    }
+};
+
+Wallet.prototype.clear = function () {
+    for (var i = 0; i < this.length; ++i) {
+        delete this[this[i].address];
+        this[i].address = null;
+        this[i].privateKey = null;
+        delete this[i];
+    }
+    this.length = 0;
+}
+
+Wallet.prototype.encrypt = function (password) {
+    var accounts = [];
```

Comment:
The format and encryption should be the same as the geth keystore. See ~/Library/Ethereum/keystore
The go code is here i think: https://github.com/ethereum/go-ethereum/blob/master/accounts/keystore/keystore.go